### PR TITLE
SEC-1 Microprofile JWT Auth - Security Connector

### PR DESCRIPTION
**Pull request by jGauravGupta**
Microprofile JWT Auth - Security Connector:
   groupId: **fish.payara.security.connectors.jwt.auth**
   artifactId: **microprofile-jwt-auth-security-connector**

Moved from https://github.com/payara/Payara/tree/master/appserver/payara-appserver-modules/microprofile/jwt-auth.

### CODE REVIEW

File: jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/cdi/CdiExtension.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 2017 Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (MarkWareham) [inline jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/cdi/CdiExtension.java]: Was this class moved or is the year a copy-paste error?

Comment (jGauravGupta) [reply jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/cdi/CdiExtension.java]: All of the classes moved from https://github.com/payara/payara/ repository.

**Comment by pdudits** [comment]
I feel the artifact coordinates are too complicated. I don't see the full picture of what the different security connectors will be and what artifacts they will contain, but I'd suggest this:

groupId: `fish.payara.microprofile.security`
artifactId: `security-connector-jwt`

This is based on assumptions:
* that these are only relevant as dependencies in microprofile applications (otherwise different root groupId would be chosen). I can imagine e. g. other config sources in microprofile.config.
* connectors are bulk of microprofile security features we expect to exist (otherwise I'd suggest security.connectors)
* there is usually not more than two artifacts per connector - having groupId per type is then superflous

**Comment by arjantijms** [comment]
> that these are only relevant as dependencies in microprofile applications

This connector is Jakarta EE Security based, and can be used everywhere that this is supported, including Tomcat. See https://blog.payara.fish/using-jakarta-security-on-tomcat

There is a MicroProfile Config dependency, although for this connector even that can be optional (it was added in 1.1 and the proprietary config option is still there too).

**Comment by jGauravGupta** [comment]
Closing as MP-JWT connector is related to the payara-mp-impl repo.

**Pull request closed without merging**